File: ecommerce-fe/src/services/chatService.js
import axiosClient from '../api/axiosClient'

export const chatService = {
  getThreads: async () => {
    const res = await axiosClient.get('/api/v1/chat/threads')
    return res.data
  },

  getOrCreateSupportThread: async () => {
    const res = await axiosClient.post('/api/v1/chat/threads/support')
    return res.data
  },

  getOrCreateShopThread: async (shopId) => {
    const res = await axiosClient.post(`/api/v1/chat/threads/shop/${shopId}`)
    return res.data
  },

  getThreadById: async (threadId) => {
    const res = await axiosClient.get(`/api/v1/chat/threads/${threadId}`)
    return res.data
  },

  getMessages: async (threadId, page = 0, size = 30) => {
    const res = await axiosClient.get(`/api/v1/chat/threads/${threadId}/messages`, {
      params: { page, size },
    })
    return res.data
  },

  sendMessage: async (threadId, content, recipientId = null) => {
    const res = await axiosClient.post(`/api/v1/chat/threads/${threadId}/messages`, {
      content,
      recipientId,
    })
    return res.data
  },

  sendImage: async (threadId, file) => {
    const formData = new FormData()
    formData.append('file', file)
    const res = await axiosClient.post(`/api/v1/chat/threads/${threadId}/images`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
    return res.data
  },

  sendVideo: async (threadId, file) => {
    const formData = new FormData()
    formData.append('file', file)
    const res = await axiosClient.post(`/api/v1/chat/threads/${threadId}/videos`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
    return res.data
  },

  editMessage: async (messageId, content) => {
    const res = await axiosClient.patch(`/api/v1/chat/messages/${messageId}`, { content })
    return res.data
  },

  deleteMessage: async (messageId) => {
    await axiosClient.delete(`/api/v1/chat/messages/${messageId}`)
  },

  markRead: async (threadId) => {
    const res = await axiosClient.put(`/api/v1/chat/threads/${threadId}/read`)
    return res.data
  },

  closeThread: async (threadId) => {
    const res = await axiosClient.put(`/api/v1/chat/threads/${threadId}/close`)
    return res.data
  },
}

export default chatService

File: ecommerce/src/main/java/com/marketplace/ecommerce/chat/service/ChatService.java
package com.marketplace.ecommerce.chat.service;

import com.marketplace.ecommerce.chat.dto.ChatMessageResponse;
import com.marketplace.ecommerce.chat.dto.ChatThreadResponse;
import com.marketplace.ecommerce.chat.dto.SendMessageRequest;
import com.marketplace.ecommerce.common.CurrentUserInfo;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.UUID;

public interface ChatService {

    List<ChatThreadResponse> getThreads(CurrentUserInfo principal);
    List<ChatThreadResponse> getThreads(CurrentUserInfo principal, String type);

    ChatThreadResponse getOrCreateSupportThread(CurrentUserInfo principal);

    ChatThreadResponse getOrCreateShopThread(CurrentUserInfo principal, UUID shopId);
    ChatThreadResponse getOrCreateShopThread(CurrentUserInfo principal, String shopIdStr);

    ChatThreadResponse getThreadById(CurrentUserInfo principal, UUID threadId);

    Page<ChatMessageResponse> getMessages(CurrentUserInfo principal, UUID threadId, int page, int size);

    ChatMessageResponse sendMessage(CurrentUserInfo principal, SendMessageRequest request);

    ChatMessageResponse sendImageMessage(CurrentUserInfo principal, UUID threadId, MultipartFile file);

    ChatMessageResponse sendVideoMessage(CurrentUserInfo principal, UUID threadId, MultipartFile file);

    ChatMessageResponse editMessage(CurrentUserInfo principal, UUID messageId, String newContent);

    void deleteMessage(CurrentUserInfo principal, UUID messageId);

    void markRead(CurrentUserInfo principal, UUID threadId);

    void closeThread(CurrentUserInfo principal, UUID threadId);

    void handleTyping(CurrentUserInfo principal, UUID threadId, boolean isTyping);
}
